src/main/java/Duke/Parser.java

```diff
@@ -0,0 +1,316 @@
+package duke;
+
+import duke.exception.DeleteOutOfRangeException;
+import duke.exception.DoneOutOfRangeException;
+import duke.exception.EmptyDeadlineException;
+import duke.exception.EmptyEventException;
+import duke.exception.EmptyTodoException;
+import duke.exception.MissingDeadlineDateException;
+import duke.exception.MissingDeleteArgumentException;
+import duke.exception.MissingDoneArgumentException;
+import duke.exception.MissingEventDateException;
+import duke.exception.MissingFindArgumentException;
+import duke.exception.UnknownCommandException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.ToDo;
+
+/**
+ * Static class that parses user inputs and executes them.
+ */
+public class Parser {
+
+    /**
+     * Parses and executes user input, adding tasks to given TaskList and output
+     * messages through given Ui.
+     *
+     * @param input  A String of user input.
+     * @param tasks  A TaskList object that the tasks will be modified in.
+     * @param ui  An Ui object that will display the outputs and collect user inputs.
+     * @return  A Boolean of whether TaskList was updated.
+     * @throws MissingDoneArgumentException  If done was input without an argument.
+     * @throws DoneOutOfRangeException  If done was input with an argument out of range.
+     * @throws MissingDeleteArgumentException  If delete was input without an argument.
+     * @throws DeleteOutOfRangeException  If delete was input with an argument out of range.
+     * @throws EmptyTodoException  If todo was input without a description.
+     * @throws MissingDeadlineDateException  If deadline was input without a date.
+     * @throws EmptyDeadlineException  If deadline was input without a description.
+     * @throws MissingEventDateException  If event was input without a date.
+     * @throws EmptyEventException  If event was input without a description.
+     * @throws UnknownCommandException  If input is not recognised by Duke.
+     * @throws MissingFindArgumentException  If find was input without a keyword.
```

Comment:
Refactoring your exceptions in the way I mentioned above will also make your docstrings cleaner